src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class Storage {
+
+    private static boolean appendMode = true;
+    private static String filePath = "./data/dukeData.txt";
+
+    /**
+     * loads data about the task list from dukeData.txt.
+     * @return an ArrayList of TaskItems that will be used to create Duke's TaskList.
+     */
+    public static ArrayList<TaskItem> loadData() {
+        File folder = new File("./data/");
+
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+        File dukeData = new File("./data/dukeData.txt");
+        try {
+            if (dukeData.createNewFile()) {
+                //System.out.println("File is created!");
+            } else {
+                //System.out.println("Data file already exists.");
+            }
+
+            if (dukeData.length() == 0) {
+                return new ArrayList<TaskItem>();
+            } else {
+                BufferedReader bufReader = new BufferedReader(new FileReader("./data/dukeData.txt"));
+                ArrayList<TaskItem> taskItems = new ArrayList<>();
+                String task = bufReader.readLine();
+                while (task != null) {
+                    String completed = task.substring(4, 5);
+                    String taskType = task.substring(1, 2);
+
+                    if (taskType.equals("D")) {
```

Comment:
This section where you check for each type of task and add them to the list could be abstracted away and placed in the `TaskItem` class or your `TaskList` class to reduce bulk in this method.

Reply:
Yup, I agree with this :)